conference.js

```diff
@@ -1343,10 +1345,35 @@ export default {
                     this.isSharingScreen = false;
                 }
                 this.setVideoMuteStatus(this.isLocalVideoMuted());
-                APP.UI.updateDesktopSharingButtons();
+                this._maybeUpdateScreenSharingStatus(oldStream, newStream);
             });
     },
 
+    /**
+     * Compares previous and provided video stream types.
+     * If desktop sharing was stopped/started it updates desktop sharing
+     * buttons in UI and notifies external application.
+     *
+     * @param {JitsiLocalTrack} [oldStream] previously used stream or null
+     * @param {JitsiLocalTrack} [newStream] new stream to use or null
+     * @private
+     * @returns {void}
+     */
+    _maybeUpdateScreenSharingStatus(oldStream, newStream) {
+        const oldVideoType = oldStream && oldStream.videoType;
+        const newVideoType = newStream && newStream.videoType;
+
+        const sharingStatusChanged = oldVideoType !== newVideoType
+            && (oldVideoType === 'desktop' || newVideoType === 'desktop');
+
+        if (sharingStatusChanged) {
+            APP.UI.updateDesktopSharingButtons();
+            APP.API.notifyScreenSharingStatusChanged(
+                this.isSharingScreen
+            );
```

Comment:
I like it. It's clear.

I want your opinion on this (and I am not requesting a change): what do you think of the approach of using a variable to store the prior value of this.isSharingScreen and seeing if this.isSharingScreen has changed instead of comparing stream types? Or a helper for setting this.isSharingScreen and emitting a change in the helper?

Reply:
I've took a look at the code and I think relying on `this.isSharingScreen` can make the code even cleaner. Saving `oldStream` to a local variable won't be necessary in this case. And instead of `_maybeUpdateScreenSharingStatus` method, there will be just a helper for setting `isSharingScreen`. Moreover, it will allow encapsulating screen sharing related code in one method, instead of spreading it between `useVideoStream` and `_maybeUpdateScreenSharingStatus`.

I think a helper can be as follows:
```es6
_setSharingScreen(newStream) {
  const wasSharingScreen = this.isSharingScreen;
  this.isSharingScreen = newStream && newStream.videoType === 'desktop';

  if (wasSharingScreen !== this.isSharingScreen) {
    APP.UI.updateDesktopSharingButtons();
    APP.API.notifyScreenSharingStatusChanged(this.isSharingScreen);
  }
}
```


then callback in `useVideoStream` will be just:
```es6
this.localVideo = newStream;
this._setSharingScreen(newStream);
if (newStream) APP.UI.addLocalStream(newStream);
this.setVideoMuteStatus(this.isLocalVideoMuted());
```

@virtuacoplenny let me know if you'd like me to implement this approach

Reply:
Yeah, that sounds good to me; please try implementing that approach. I think it'd be a good precedent to go with what is believed to be the cleaner implementation.

This review has been going on for a bit so when the change is ready, I'll give your branch a re-review and a manual test (for sanity) and merge myself.